Runtime/Extensions/CollectionsExtensions.cs

```diff
@@ -127,17 +127,44 @@ public static T[] ExportDictionaryValuesAsArray<T>(this Dictionary<uint, T> inpu
         }
        
         /// <summary>
-        /// Validate if a list contains an item and add it if not found
+        /// Validate if a list contains an item and add it if not found.
         /// </summary>
-        /// <typeparam name="T">Data type used in the List</typeparam>
-        /// <param name="list">The instance of the List to validate</param>
-        /// <param name="item"></param>
+        /// <typeparam name="T">Data type used in the List.</typeparam>
+        /// <param name="list">The instance of the List to validate.</param>
+        /// <param name="item">The item of Type T to add to the list if not found</param>
         public static void EnsureListItem<T>(this IList<T> list, T item)
         {
             if (!list.Contains(item))
             {
                 list.Add(item);
             }
         }
+
+        /// <summary>
+        /// Validate if a <see cref="Dictionary{TKey, TValue}"/> contains an item and add it if not found."/>
+        /// </summary>
+        /// <remarks>
+        /// Will update the existing value when found unless overridden using the 'update' parameter
+        /// </remarks>
+        /// <typeparam name="TKey">Data type used in the Dictionary Key.</typeparam>
+        /// <typeparam name="TValue">Data type used in the Dictionary Value.</typeparam>
+        /// <param name="dictionary">The instance of the <see cref="Dictionary{TKey, TValue}"/> to validate.</param>
+        /// <param name="key">The Key of a <see cref="KeyValuePair{TKey, TValue}"/> to validate against the dictionary with.</param>
+        /// <param name="value">The Value of a <see cref="KeyValuePair{TKey, TValue}"/> to set the dictionary item with if required.</param>
+        /// <param name="update">By default, the Ensure function will override the existing dictionary value if found, if this is not required it can be overridden with this bool.  Setting this to <see cref="false"/> will leave the dictionary item untouched if found.</param>
+        public static void EnsureDictionaryItem<TKey, TValue>(this Dictionary<TKey, TValue> dictionary, TKey key, TValue value, bool update = true)
+        {
+            if(!dictionary.TryGetValue(key, out _))
```

Comment:
Actually no, because if you leave it at the default "update = true", then the item would be added to the dictionary if not found and then UPDATED again.

In its current state, it would ONLY be updated if found and only added if not found.

Reply:
It wouldn't because there is a return statement

Reply:
Didn't spot that and that is a fair point.  Will update